internal/ls/completions.go

```diff
@@ -2982,11 +2983,11 @@ func getCompletionsSymbolKind(kind ScriptElementKind) lsproto.CompletionItemKind
 // by the language service consistent with what TS Server does and what editors typically do. This also makes
 // completions tests make more sense. We used to sort only alphabetically and only in the server layer, but
 // this made tests really weird, since most fourslash tests don't use the server.
-func compareCompletionEntries(entryInSlice *lsproto.CompletionItem, entryToInsert *lsproto.CompletionItem) int {
-	// !!! use locale-aware comparison
-	result := stringutil.CompareStringsCaseSensitive(*entryInSlice.SortText, *entryToInsert.SortText)
+func (l *LanguageService) compareCompletionEntries(entryInSlice *lsproto.CompletionItem, entryToInsert *lsproto.CompletionItem) int {
+	compareStrings := collate.New(l.host.GetLocale()).CompareString
```

Comment:
I suspect we'll want to avoid calling `New` often, since that's not a cheap operation.

Probably, we should make some sort of new `locale` package (or dump it in core) which has a struct that caches the collator and so on.

Reply:
Actually reading their impl, I'm not sure collators are concurrency safe 😓

Reply:
Yeah, sorry, I missed the description and clicked right to the code in my email.